hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -800,6 +805,24 @@ public void notifyLogFailed(Throwable t, LogEntryProto failedEntry) {
     return future;
   }
 
+  @Override
+  public void notifyGroupRemove() {
+    ratisServer.notifyGroupRemove(gid);
+    // Make best effort to quasi-close all the containers on group removal.
+    // Containers already in terminal state like CLOSED or UNHEALTHY will not
+    // be affected.
+    for (Long cid : createContainerSet) {
+      try {
+        containerController.markContainerForClose(cid);
+      } catch (IOException e) {
+      }
+      try {
+        containerController.quasiCloseContainer(cid);
+      } catch (IOException e) {
+      }
+    }
+  }
+
```

Comment:
If markContainerForClose fails, quasiCloseContainer will definitely fail. We can put both of the calls into same try catch.

Reply:
addressed in 2nd comit